crates/project/src/environment.rs

```diff
@@ -165,59 +181,40 @@ impl From<EnvironmentOrigin> for String {
     }
 }
 
+pub struct EnvironmentErrorMessage(pub String);
+
 #[cfg(any(test, feature = "test-support"))]
 async fn load_shell_environment(
     _dir: &Path,
     _load_direnv: &DirenvSettings,
-) -> Result<HashMap<String, String>> {
-    Ok([("ZED_FAKE_TEST_ENV".into(), "true".into())]
+) -> Result<(HashMap<String, String>, Option<EnvironmentErrorMessage>)> {
+    let fake_env = [("ZED_FAKE_TEST_ENV".into(), "true".into())]
         .into_iter()
-        .collect())
+        .collect();
+    Ok((fake_env, None))
 }
 
 #[cfg(not(any(test, feature = "test-support")))]
 async fn load_shell_environment(
     dir: &Path,
     load_direnv: &DirenvSettings,
-) -> Result<HashMap<String, String>> {
+) -> Result<(HashMap<String, String>, Option<EnvironmentErrorMessage>)> {
```

Comment:
By introducing this new type instead of just using `Result`, we now have two types of errors, one that gets displayed and one that doesn't. But we have two anyhow calls in there that can return an error — why not display this too? I think the whole code could be simpler if you just return `Result<Hashmap<String, String>>` here and save the errors above.

It also makes the code less confusing above, where there's now a `log_err()` but also an optional error.

Reply:
The problem is, direnv errors don't prevent us from continuing to retrieve at least the login environment, so they are not exactly errors, more like warnings, which we could return besides the retrieved environment. Anyhow errors, on the other hand, terminate the whole process of getting the environment

Reply:
Let's consider a couple of cases:
1. `direnv` call fails, call to the login shell succeeds, we return `(Some(env), Some(direnv_error))`
2. spawning the login shell fails, the `env` output is mailformed, or something like that, we return `(None, Some(message)`
3. Everything succeeds, we return `(Some(env), None)`